mm2src/mm2_main/src/lp_network.rs

```diff
@@ -138,36 +139,68 @@ async fn process_p2p_message(
     mut message: GossipsubMessage,
     i_am_relay: bool,
 ) {
+    fn is_valid(topics: &Vec<TopicHash>) -> bool {
+        if topics.is_empty() {
+            return false;
+        }
+
+        let first_topic_prefix = topics[0].as_str().split(TOPIC_SEPARATOR).next().unwrap_or_default();
+        for item in topics.iter().skip(1) {
+            if !item.as_str().starts_with(first_topic_prefix) {
+                return false;
+            }
+        }
+
+        true
+    }
+
     let mut to_propagate = false;
     let mut orderbook_pairs = vec![];
 
     message.topics.dedup();
     drop_mutability!(message);
 
+    if !is_valid(&message.topics) {
+        return;
+    }
+
     for topic in message.topics {
         let mut split = topic.as_str().split(TOPIC_SEPARATOR);
         match split.next() {
             Some(lp_ordermatch::ORDERBOOK_PREFIX) => {
                 if let Some(pair) = split.next() {
+                    if decode_signed::<OrdermatchMessage>(&message.data).is_err() {
```

Comment:
I agree on the orderbook handling, it seems completely wrong. Will move it from `process_p2p_message`

> decode_signed doesn't log any message, and nobody will know that an error occurred because we exit silently. It would be also more clear if process_p2p_message returned an Result we could use map_err and ? syntax together.

`process_p2p_message` is continuously spawned on the threads inside of a loop. It's missing just a couple places to be logged, which I will add debug logs.

Reply:
Is this resolved yet?